File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnTertiaryLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl34
            AppTheme.STANDARD -> R.color.cstl34
            AppTheme.BLUE -> R.color.cbl34
            AppTheme.GREEN -> R.color.cgl34
            AppTheme.RED -> R.color.crl34
            AppTheme.NOTHING -> R.color.cnol34
        }
    }
}

File: lib/layer2_core/light/e_on_tertiary_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnTertiaryLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl34;
      case 'standard': return cstl34;
      case 'blue': return cbl34;
      case 'green': return cgl34;
      case 'red': return crl34;
      case 'nothing': return cnol34;
      default: return cstl34;
    }
  }
}
